File: HW 11/Java/Floodfill.java
public class FloodFill {
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        if (image[sr][sc] == color) {
            return image;
        }

        int originalColor = image[sr][sc];
        dfs(image, sr, sc, originalColor, color);
        return image;
    }

    private void dfs(int[][] image, int r, int c, int originalColor, int newColor) {
        if (r < 0 || r >= image.length || c < 0 || c >= image[0].length || image[r][c] != originalColor) {
            return;
        }

        image[r][c] = newColor;
        dfs(image, r + 1, c, originalColor, newColor);
        dfs(image, r - 1, c, originalColor, newColor);
        dfs(image, r, c + 1, originalColor, newColor);
        dfs(image, r, c - 1, originalColor, newColor);
    }
}

File: HW 11/Python/floodfill.py
def floodFill(image, sr, sc, color):
    if image[sr][sc] == color:
        return image
    
    def dfs(image, r, c, color, new_color):
        if r < 0 or r >= len(image) or c < 0 or c >= len(image[0]) or image[r][c] != color:
            return
        
        image[r][c] = new_color
        dfs(image, r+1, c, color, new_color)
        dfs(image, r-1, c, color, new_color)
        dfs(image, r, c+1, color, new_color)
        dfs(image, r, c-1, color, new_color)
    
    dfs(image, sr, sc, image[sr][sc], color)
    return image
